Android/app/src/main/java/com/example/androidgui/inventoryactivities/AddInventoryActivity.java

```diff
@@ -36,14 +41,22 @@ public void onClick(View v) {
         String iamount = amount.getText().toString();
         String idate = date.getText().toString();
         String ifreshness = freshness.getText().toString();
+        String[] paras = this.updateInventoryList(iname, iprice, iamount, idate, ifreshness);
+        im.addNewInventory(paras);
+    }
+    @Override
+    public String[] updateInventoryList(String name, String price, String amount, String date, String freshness){
         String para;
-        if(ifreshness.equals("N/A")){
-            para = iname+","+iprice+","+iamount+","+idate;
+        if(freshness.equals("N/A")){
```

Comment:
the presenter should be doing this. So have the activity call a method in the presenter with these parameters, then have the presenter do this stuff.